src/persist-client/src/usage.rs

```diff
@@ -44,6 +45,35 @@ impl StorageUsageClient {
         .await
     }
 
+    /// Returns a map of [crate::ShardId] to its size (in bytes) stored in persist,
+    /// and the total bytes stored for unattributable data (i.e. does not have a shard
+    /// key in its name). This latter count _should_ always be 0, so being nonzero
+    /// indicates either persist wrote an invalid blob key, or another process is
+    /// storing data under the same path (!)
+    pub async fn shard_sizes(&self) -> (HashMap<ShardId, u64>, u64) {
```

Comment:
it feels more natural to me for the map key to be `Option<ShardId>` and then use `None` for the unattributable data, but happy to defer to you or @jpepin